drivers/serial/uart_nrf_sw_lpuart.c

```diff
@@ -591,17 +832,59 @@ static int lpuart_init(struct device *dev)
 	k_timer_init(&data->tx_timer, tx_timeout, NULL);
 	k_timer_user_data_set(&data->tx_timer, dev);
 
-	return uart_callback_set(data->uart, uart_callback, (void *)dev);
+	err = uart_callback_set(data->uart, uart_callback, (void *)dev);
+	if (err < 0) {
+		return -EINVAL;
+	}
+
+#if CONFIG_NRF_SW_LPUART_INT_DRIVEN
+	err = uart_callback_set(dev, int_driven_evt_handler, dev);
```

Comment:
Passing `dev` as `user_data` is unnecessary in this case, as the device pointer is the same, so I'd suggest to change it to `NULL` so there is one less update needed during the next upmerge (pointers to devices got "constified" upstream).

Reply:
right, it comes back to the times when callback did not have `device` passed. fixed.